src/ssl.c

```diff
@@ -1070,6 +1115,20 @@ void ssl_setup_session_resumption(SSL_CTX **contexts, size_t num_contexts, struc
 #endif
 }
 
+void ssl_setup_session_resumption_ptls(ptls_context_t *ptls, quicly_context_t *quic)
+{
+    if (conf.ticket.update_thread != NULL) {
+        struct encrypt_ticket_ptls_t *encryptor = malloc(sizeof(*encryptor));
+        *encryptor = (struct encrypt_ticket_ptls_t){{encrypt_ticket_ptls}};
```

Comment:
I'm not familiar with this syntax. What does this do?

Reply:
I'd imagine it's constructing a variable of type `struct encrypt_ticket_ptls_t`, whose first member being `{encrypt_ticket_ptls}` and the rest being zero, and assign it to `*encryptor`.

So it's an alias of
```
struct encrypt_ticket_ptls_t tmp = {{encrypt_ticket_ptls}};
*encryptor = tmp;
```

Reply:
And what is `encrypt_ticket_ptls` ?

Reply:
That's the callback function.

Line 1122 initializes a callback object, setting the callback function to `encrypt_ticket_ptls` as well as implicitly initializing all the other fields to zero (or NULL, depending on their types).